src/System.Private.ServiceModel/tools/CertificateGenerator/CertificateManager.cs

```diff
@@ -186,6 +199,16 @@ public static X509Certificate2 CreateAndInstallLocalMachineCertificates(Certific
                 InstallCertificateToRootStore(rootCertificate);
                 InstallCertificateToMyStore(hostCert, certificateCreationSettings.ValidityType == CertificateValidityType.Valid);
                 s_localCertificate = hostCert;
+
+                // Create the PeerTrust cert
+                certificateCreationSettings = new CertificateCreationSettings()
+                {
+                    FriendlyName = "WCF Bridge - UserPeerTrustCertificateResource",
+                    Subject = fqdn,
+                    SubjectAlternativeNames = new string[] { fqdn, hostname, "localhost" }
+                };
+                var peerCert = certificateGenerator.CreateMachineCertificate(certificateCreationSettings).Certificate;
+                InstallCertificateToTrustedPeopleStore(peerCert, certificateCreationSettings.ValidityType == CertificateValidityType.Valid);
             }
```

Comment:
If we go there (which we can discuss), you're right we will need a new cert, a new REST method to get it, a new ConditionalFact to detect and obtain it, a new TestProperty, etc.   And if we go there for Https, we'll need to update build scripts, etc.  I guess the question to ask (though hard to answer) is "What new WCF product code paths would be tested by this kind of cert?"

---
In reply to: [71627543](https://github.com/dotnet/wcf/pull/1382#discussion_r71627543) [](ancestors = 71627543)